pkg/controllers/jobset_controller.go

```diff
@@ -219,6 +222,63 @@ func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet)
 	return &ownedJobs, nil
 }
 
+func (r *JobSetReconciler) calculateAndUpdateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Prepare replicatedJobsReady for optimal iteration
+	replicatedJobsReady := map[string]int32{}
+	for _, replicatedJob := range js.Spec.ReplicatedJobs {
+		replicatedJobsReady[replicatedJob.Name] = 0
+	}
+
+	// Calculate ReadyPods for each Replicated Job
+	for _, job := range append(jobs.active, append(jobs.successful, jobs.failed...)...) {
+		ready := pointer.Int32Deref(job.Status.Ready, 0)
+		// parallelism is always set as it is otherwise defaulted by k8s to 1
+		podsCount := *(job.Spec.Parallelism)
+		if job.Spec.Completions != nil && *job.Spec.Completions < podsCount {
+			podsCount = *job.Spec.Completions
+		}
+		if job.Status.Succeeded+ready >= podsCount {
```

Comment:
```suggestion
		if job.Status.Succeeded + ready >= podsCount {
```

Reply:
![image](https://github.com/kubernetes-sigs/jobset/assets/14259249/4cca6548-779d-43d9-83fe-706cde1cd99d)

It is a gofmt standard in if statements.

Reply:
Oh interesting I didn't know that, I find it more readable with the spaces.. Oh well, you can disregard this comment then :)